packages/rest/src/resource.ts

```diff
@@ -118,7 +118,6 @@ function cloneKeyProperties(context: DecoratorContext, target: Model, resourceTy
       decorators,
       optional: false,
       model: target,
-      sourceProperty: keyProperty,
```

Comment:
I wonder if this needs to be explicitly unset sourceProperty so we don't inherit it?